src/main/java/com/minecolonies/colony/permissions/Permissions.java

```diff
@@ -582,43 +617,43 @@ public boolean isDirty()
     }
 
     /**
-     * Marks instance not dirty
+     * Marks instance not dirty.
      */
     public void clearDirty()
     {
         dirty = false;
     }
 
     /**
-     * Serializes network data
+     * Serializes network data.
      *
-     * @param buf        {@link ByteBuf} to write to
-     * @param viewerRank Rank of the viewer
+     * @param buf        {@link ByteBuf} to write to.
+     * @param viewerRank Rank of the viewer.
      */
-    public void serializeViewNetworkData(@NotNull ByteBuf buf, @NotNull Permissions.Rank viewerRank)
+    public void serializeViewNetworkData(@NotNull final ByteBuf buf, @NotNull final Permissions.Rank viewerRank)
     {
         ByteBufUtils.writeUTF8String(buf, viewerRank.name());
 
         //  Owners
         buf.writeInt(players.size());
-        for (@NotNull Map.Entry<UUID, Player> player : players.entrySet())
+        for (@NotNull final Map.Entry<UUID, Player> player : players.entrySet())
         {
             PacketUtils.writeUUID(buf, player.getKey());
             ByteBufUtils.writeUTF8String(buf, player.getValue().name);
             ByteBufUtils.writeUTF8String(buf, player.getValue().rank.name());
         }
 
         // Permissions
-        buf.writeInt(permissions.size());
-        for (@NotNull Map.Entry<Rank, Integer> entry : permissions.entrySet())
+        buf.writeInt(permissionMap.size());
+        for (@NotNull final Map.Entry<Rank, Integer> entry : permissionMap.entrySet())
         {
             ByteBufUtils.writeUTF8String(buf, entry.getKey().name());
             buf.writeInt(entry.getValue());
         }
     }
 
     /**
-     * Ranks within a colony
+     * Ranks within a colony.
      */
     public enum Rank
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this class from 28 to the maximum allowed 25 or externalize it in a public class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2972)